### ISSUE TITLE
Bug when no frameMax parameter is specified to Loader.spritesheet

### ISSUE THREAD

**Issue by Lucas-C** [photonstorm/phaser-ce#448]
This bug occurs only from the 1680th frame in the spritesheet

This is a bug in the API:
  * **Phaser version(s):** 2.6.2
  * **Live example:** https://chezsoi.org/lucas/phaserjs_spritesheet_bug_repro.html?1767
  * **What should happen:** without the `?1767` query string fixing the `frameMax` parameter to `Loader.spritesheet`, the same sprite should be displayed
  * **What happens instead:** when no `frameMax` is passed to `Loader.spritesheet` (default value is `-1`), the first sprite of the spritesheet is displayed instead. But all is fine if we try to render the sprite before (1679th one)

There is the code used in the live example:
```javascript
window.onload = function () {
  var game = new Phaser.Game(800, 600, /*renderer=*/Phaser.AUTO, /*parent=*/'',
                             /*state=*/{preload: preload, create: create},
                             /*transparent=*/false, /*antialias=*/false)

  function preload() {
    // Test with ?1767 query string
    var frameMax = window.location.search ? +window.location.search.substring(1) : -1
    game.load.spritesheet('MySpritesheet', './roguelikeSheet_transparent.png', 16, 16,
                          /*frameMax=*/frameMax, /*margin=*/0, /*space=*/1)
  }

  function create() {
    // The following will render the first tile if loaded with frameMax=-1, whereas 1679 is OK
    var sprite = game.add.sprite(game.world.centerX, game.world.centerY, 'MySpritesheet', 1680)
    sprite.anchor.setTo(0.5)
    sprite.scale.setTo(4)
  }
}
```

The API function called is [`Loader.spritesheet`](https://github.com/photonstorm/phaser/blob/v2.6.2/src/loader/Loader.js#L981), but I think the issue relies in [`AnimationParser.spriteSheet`](https://github.com/photonstorm/phaser/blob/v2.6.2/src/animation/AnimationParser.js#L28)

**Comment by Lucas-C** [comment]
The spritesheet loaded has 57x31 sprites of size 16 pixels.
However `AnimationParser.spriteSheet` code only counts 56 rows x 30 columns (hence the 1680 number)

I think the code should be:
```
var row = Math.floor((width - margin + spacing) / (frameWidth + spacing));
var column = Math.floor((height - margin + spacing) / (frameHeight + spacing));
```

**Comment by samme** [comment]
> game.load.spritesheet('MySpritesheet', './roguelikeSheet_transparent.png', 16, 16, /*frameMax=*/frameMax, /*margin=*/0, /*space=*/1)

Does this work as expected if you use `margin = -1`, `space = 1`?

**Comment by Lucas-C** [comment]
Yes, this would give the correct result result mathematically:
```
var row = Math.floor((width - margin) / (frameWidth + spacing))
        = Math.floor((985 - (-1)) / (16 + 1))
        = Math.floor(56) = 56
var row = Math.floor((width - margin) / (frameWidth + spacing))
        = Math.floor((526 - (-1)) / (16 + 1))
        = Math.floor(31) = 31
```

However, this is still a bug !

Passing `-1` as margin is **hackish**.
The reason why I added the `+ spacing` in the formula is **geometrical**.

**Comment by samme** [comment]
I think `spacing` implies the same length on the exterior edges as well. If that's absent, you do need a negative `margin`.

The definition is unclear, and it may not be the best one, but since it's always worked this way, I'd rather just change the documentation to explain it better. I'm open to feedback though.

#### margin = -1, spacing = 1

![tiles with margin = -1, spacing = 1](https://user-images.githubusercontent.com/435681/35877976-b8e4de4a-0b2b-11e8-9b7e-391f0753c55c.png)

#### margin = 0, spacing = 1

![tiles with margin = 0, spacing = 1](https://user-images.githubusercontent.com/435681/35877977-b900dd02-0b2b-11e8-90f1-bcd35d628ff9.png)

#### margin = 1, spacing = 1

![tiles with margin = 1, spacing = 1](https://user-images.githubusercontent.com/435681/35877978-b919eebe-0b2b-11e8-805f-c8f3e046d67a.png)

**Comment by Lucas-C** [comment]
I dislike the negative margin concept, but I don't have any strong argument against simply clarifying this in the documentation.

**Comment by samme** [comment]
So according to the code,

    expected width = margin + rows * (frameWidth + spacing)

    expected height = margin + cols * (frameHeight + spacing)

So that would give a layout something like

    5px : mfsfs     margin=1 spacing=1
    7px : mfssfss   margin=1 spacing=2
    9px : mfsssfsss margin=1 spacing=3

**Comment by samme** [comment]
Corrected the docs (3fd72cc). Thanks @Lucas-C
